client/table.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package client
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/proto"
+)
+
+// Row ...
+type Row []interface{}
+
+// IndexKey ...
+type IndexKey []interface{}
+
+// Index ...
+type Index struct {
+}
+
+// Scan retrieves 1 or more columns from the rows between start (inclusive) and
+// end (exclusive). If columns is empty all of the columns for the table are
+// retrieved.
+func (i *Index) Scan(txn *Txn, start, end Row, maxRows int, columns ...string) ([]Row, error) {
+	return nil, nil
+}
+
+// Key constructs an IndexKey from the specified column values.
+func (i *Index) Key(values ...interface{}) IndexKey {
+	// TODO(pmattis): Check that "values" contains the same number of elements as
+	// the index.
+	return IndexKey(values)
+}
+
+// Table ...
+type Table struct {
+	// Schema ...
+	Columns []string
+	PK      *Index
+	indexes map[string]*Index
+}
+
+// OpenTable ...
```

Comment:
Why would you need `TableSchema` here? I was imagining that opening the table would retrieve the schema from the server. Not sure if that is necessary, though. We could do with loading the schema here and just let the server do the validation which it is going to have to do anyways. Thoughts?